kclvm/tools/src/LSP/src/hover.rs

```diff
@@ -0,0 +1,60 @@
+use indexmap::IndexSet;
+use kclvm_ast::ast::Program;
+use kclvm_error::Position as KCLPos;
+use kclvm_sema::resolver::scope::{ProgramScope, ScopeObjectKind};
+use lsp_types::{Hover, HoverContents, MarkedString};
+
+use crate::goto_def::find_definition_objs;
+
+/// Returns a short text describing element at position.
+/// Specifically, the doc for schema and schema attr(todo)
+pub(crate) fn hover(
+    program: &Program,
+    kcl_pos: &KCLPos,
+    prog_scope: &ProgramScope,
+) -> Option<lsp_types::Hover> {
+    match program.pos_to_stmt(kcl_pos) {
+        Some(node) => match node.node {
+            _ => {
+                let objs = find_definition_objs(node, kcl_pos, prog_scope);
+                let mut docs: IndexSet<String> = IndexSet::new();
+                for obj in &objs {
+                    match obj.kind {
+                        ScopeObjectKind::Definition => {
+                            docs.insert(obj.ty.ty_str());
+                            let doc = obj.ty.into_schema_type().doc.clone();
+                            if doc != "" {
```

Comment:
`!doc.is_empty()` U can run `cargo clippy --fix --allow-dirty` and `cargo fmt` to fix this issue in the LSP crate.